implementations/c/ockam/transport/posix_socket/socket_udp.c

```diff
@@ -181,17 +181,17 @@ ockam_error_t socket_udp_read(void* ctx, uint8_t* buffer, size_t buffer_size, si
 
 ockam_error_t socket_udp_write(void* ctx, uint8_t* buffer, size_t buffer_length)
 {
+  // TODO: Check buffer_length can fit in ssize_t?
   ockam_error_t     error       = OCKAM_ERROR_NONE;
   socket_udp_ctx_t* p_udp_ctx   = (socket_udp_ctx_t*) ctx;
   posix_socket_t*   p_socket    = &p_udp_ctx->posix_socket;
-  size_t            bytes_sent  = 0;
-
-  bytes_sent = sendto(p_socket->socket_fd,
-                      buffer,
-                      buffer_length,
-                      0,
-                      (struct sockaddr*) &p_socket->remote_sockaddr,
-                      sizeof(p_udp_ctx->posix_socket.remote_sockaddr));
+
+  ssize_t bytes_sent = sendto(p_socket->socket_fd,
+                              buffer,
+                              buffer_length,
+                              0,
+                              (struct sockaddr*) &p_socket->remote_sockaddr,
+                              sizeof(p_udp_ctx->posix_socket.remote_sockaddr));
   if (bytes_sent != buffer_length) {
```

Comment:
one is signed the other is unsigned. Should probably do an explicit cast and saturate.

Reply:
I think worth checking bytes_sent for negative values

Reply:
Added